testproject/testapp/tests.py

```diff
@@ -221,6 +222,17 @@ def testDelete(self):
         self.obj.delete()
         self.assertEqual(self.model.objects.count(), 0)
 
+    def testDeleteWithIn(self):
+        expected = self.create_multiple_models()
+        random.shuffle(expected)
+        delete_ids = expected[:random.randint(1, len(expected) - 1)]
+        count_rows = self.model.objects.all().count()
+        qs = self.model.objects.filter(id__in=delete_ids).delete()
+        after_count = self.model.objects.all().count()
+        self.assertEqual(count_rows - len(delete_ids), after_count)
+        list(self.model.objects.order_by())
```

Comment:
First, using random leads to test failures that are very difficult to reproduce.
Second, test should not contain lines that are excess for test subject.
And third, add assert that delete() deletes only what it should, and skips what it shouldn't.